docs/index.md

```diff
@@ -462,7 +524,9 @@ describe('Array', function() {
 });
 ```
 
-Pending tests will be included in the test results, and marked as pending. A pending test is not considered a failed test.
+By appending `.skip()`, you may also tell Mocha to ignore a test: [inclusive tests](#inclusive-tests).
```

Comment:
suggestion:

```md
By appending `.skip()`, you may also tell Mocha to [ignore a test](#inclusive-tests).
```

Reply:
wait... I'm confused now.  `this.skip()` means "test is pending", not "test is skipped"?

I think we could be more clear here.  For your consideration:

- Maybe we can, instead, make `this.skip()` _report a test as skipped_, and `this.pending()` _report a test as pending_.  The behavior would be roughly similar (the test wouldn't get run), but the test would not be displayed as "pending" _unless_ the user called `this.pending()` instead of `this.skip()` (likewise for `describe.skip()`/`describe.pending()`/`it.skip()`/`it.pending()`).  This would be _less of_ a breaking change than just trying to rename `skip()` to `pending()`.
- Maybe instead of calling them "skipped tests", as you introduce above, we can just use a different word.  "Omitted" tests, maybe?  
- This may be a good place to update the terminology we use elsewhere re: "inclusive" (`it.skip()`) and "exclusive" (`it.only()`) tests.  Those terms always felt vague to me.  Instead of "inclusive", we could say "omitted", and instead of "exclusive", we could say "selected" or "picked" or something.

Reply:
The current situation (incl. this PR) is:
- `pending`: set by the user
  - test without callback it('some test')
  - static: it.skip() or describe.skip()
  - conditional: this.skip()
- `skipped`: test not run because of a failing hook

I haven't changed anything on the first item `pending`. This mismatch between our terminology and function is an old problem. We could replace label `pending` by label `skipped`, but this would break a few third-party tools, reporters, ...
I certainly don't want to rename/add a new `this.pending()`.
And the `pending` part is out of this PR's scope. 

The second label `skipped` is new and is about failing hooks. We could rename it to "omitted" or "dropped" or whatever. Eg. Cypress and Mochawesome are using `skipped` for tests omitted by failing hooks. That's why I have chosen `skipped`. But yes, it is confusing.

We need two labels: pending (or we rename it to skipped) and a new one for hook-failed-omitted tests. So I need to know which name to chose.

Reply:
A bit of brainstorm:

My thoughts are, when I mark a test not to run, I want Mocha to ignore it, to disable it, to skip it, to omit it.

If it's skipped because of hook-failed, Mocha is skipping it, leaving out, or the test was unrun, unused, benched, called off, unprepared, dropped, withdrawn, canceled, halted. And some of this could apply to the previous case as well.

I think my favorites so far for hook-failed-omitted are `called off`, `canceled`, `halted`. I think called off even best.

It could be interesting to amass a large number of these potential labels, and run a survey through users of Mocha to see how well they feel each of those words describes (from 1 to 10) those two cases. Not to base our choice on our opinion alone.